--- a/src/libmpg123/mpg123.h.in	2019-10-26 15:06:16.000000000 +0800
+++ b/src/libmpg123/mpg123.h.in	2019-11-05 18:33:33.094028600 +0800
@@ -17,7 +17,7 @@
  * This should be incremented at least each time a new symbol is added
  * to the header.
  */
-#define MPG123_API_VERSION @API_VERSION@
+#define MPG123_API_VERSION 44
 
 #ifndef MPG123_EXPORT
 /** Defines needed for MS Visual Studio(tm) DLL builds.
